includes/functions.php

```diff
@@ -2058,39 +2060,104 @@ function tsml_scan_data_source($data_source_url)
 	}
 }
 
-//function:	Returns boolean indicator when data source changes detected
-function tsml_import_has_changes($meetings, $data_source_count_meetings, $data_source_last_refresh)
+//function:	Returns summary list of modified records when data source changes detected
+function tsml_import_has_changes($data_source_url, $meetings, $data_source_count_meetings, $data_source_last_refresh)
 {
+	$is_there_change = false;
+	$meetings_updated = $db_meetings = $feed_slugs = $message_lines = [];
+	$days = [0 => 'Sunday',1 => 'Monday',2 => 'Tuesday',3 => 'Wednesday',4 => 'Thursday',5 => 'Friday',6 => 'Saturday'];
 
-	$meetings_updated = array();
+	//get db meetings 
+	$all_db_meetings = tsml_get_meetings();
+	$ds_ids = tsml_get_data_source_ids($data_source_url);
+	sort($ds_ids);
 
-	//allow theme-defined function to reformat import - issue #439
-	/*if (function_exists('tsml_import_reformat')) {
-		$meetings = tsml_import_reformat($meetings);
-	}*/
+	/* filter out all but the data source meetings  */
+	foreach ($all_db_meetings as $db_meeting) {
+	    $db_id = $db_meeting['id'];
+		if (in_array($db_id, $ds_ids)) {
+		    array_push($db_meetings, $db_meeting);
+		}
+	}
+
+	// create array of db slugs for matching
+	$db_slugs = array_column($db_meetings, 'slug', 'id');
+	sort($db_slugs);
+	$db_slugs_cnt = count($db_slugs);
 
-	if (count($meetings) !== $data_source_count_meetings) {
+	// list changed and new meetings found in the ds feed
+	foreach ($meetings as $meeting) {
 
-		if (count($meetings) > $data_source_count_meetings) {
-			$meetings_updated[] = 'Feed Records: ' . count($meetings) . ' more than DB Records: ' . $data_source_count_meetings;
-		} elseif (count($meetings) < $data_source_count_meetings) {
-			$meetings_updated[] = 'Feed Records: ' . count($meetings) . ' less than DB Records: ' . $data_source_count_meetings;
+	    $mtg_slug =  $meeting['slug'];
+		if (is_array($meeting['day'])) {
+			$dow = implode("",  $meeting['day']);
```

Comment:
what happens if `$meeting['day']` is an array, and $dow becomes, say the string `24`? Does it show up like that in the report? feel like this could be refactored a little to support "Tuesday, Thursday"

Reply:
I felt that was pretty weak. I covered only a couple of different scenarios and didn't spend enough time thinking it through. Will put some more thought into it...